### ISSUE TITLE
can not achieve the performance which mentioned in the original paper

### ISSUE THREAD

**Issue by onlyonewater** [octavian-ganea/equidock_public#21]
Hi, @AxelGiottonini, when I use the command `CUDA_VISIBLE_DEVICES=0 python -m src.train -hyper_search
` to run the code, I get the following results:
[2023-03-09 05:47:00.038149]  [FINAL TEST for dips] --> epoch -1/10000 || mean/median complex rmsd 16.2906 / 15.7649 || mean/median ligand rmsd 35.9814 / 33.6197 || mean/median sqrt pocket OT loss 28.6057 || intersection loss 21.3417 || mean/median receptor rmsd 0.0000 / 0.0000

[2023-03-09 09:37:52.987988]  [FINAL TEST for db5] --> epoch -1/10000 || mean/median complex rmsd 16.7756 / 16.5510 || mean/median ligand rmsd 40.3175 / 36.7189 || mean/median sqrt pocket OT loss 31.0876 || intersection loss 28.0045 || mean/median receptor rmsd 0.0000 / 0.0000

from the results, we can get the mean rmsd of dips test is 16.29, and the mean rmsd of db5 test is 16.77, but in paper, table 1 shows that the mean rmsd of dips test is 14.52, and the mean rmsd of db5 is 14.72, it has a lower performance than mentioned in the original paper
![image](https://user-images.githubusercontent.com/41667733/223901418-47dd345f-6ba3-44e3-af16-70726c6a08ae.png)
 what is wrong?

and how to get the interface rmsd?

**Comment by onlyonewater** [comment]
and when I run the command `CUDA_VISIBLE_DEVICES=0 python -m src.inference_rigid`, an error has appeared: File "/nvme/xxxxx/projects/docking-task/equidock_public/src/inference_rigid.py", line 203, in main 
assert np.linalg.norm(new_residues - model_ligand_coors_deform_list[0].detach().cpu().numpy()) < 1e-1
AssertionError
so, how to deal with this error?

**Comment by onlyonewater** [comment]
and when I note this line code, the code could run successfully, and I use `python -m src.test_all_methods.eval_pdb_outputset` to get the results, it shows: `For  dips  method =  equidock ; num test files = 100  complex_rmsd_CA median/mean/std =  14.964256286621094 / 15.607631420575272  +-  7.329030992174103  complex_interface_rmsd CA median/mean/std =  11.410481452941895 / 12.017205838562273  +-  5.641359052105475`, it seems that it also has a lower performance than the original paper.

**Comment by onlyonewater** [comment]
and I found that if I note this line code, using the pre-trained model you provided, and using `python -m src.test_all_methods.eval_pdb_outputset`, it shows: `For  dips  method =  equidock ; num test files = 100  complex_rmsd_CA median/mean/std =  13.295709133148193 / 14.52530690637253  +-  7.132719173736529  complex_interface_rmsd CA median/mean/std =  10.186793327331543 / 11.92203870816779  +-  7.009077590008056`, and it gets the similar results mentioned in the original paper, so I think noting this line code may be a solution for this error?

**Comment by AxelGiottonini** [comment]
Hey @onlyonewater !

For the first point, if you want to get your statistics, you need to run the src/inference_rigid.py and src/utils/eval.py script. As mentioned in other issues, I suspect the statistics to be biased for Equidock as considering the complex RMSD/iRMSD, the receptor RMSD is used but it is always equal to 0 by design of the model.

For the assertion error, I never faced it. Maybe trying to remove it could solve the problem but  I do not remember why the authors used it. So you may wan to figure out why they use this condition.

I don't really get your point on the last two questions, but I'd be happy to help you with them !

**By the way, I am not related to any of the authors of the project nor their lab. I just worked with EquiDock for a master project.**

Sincerely Meow !

**Comment by onlyonewater** [comment]
hi, @AxelGiottonini , thanks for your replying, yeah, the user should run the src/inference_rigid.py and src/test_all_methods/eval_pdb_outputset.py to get the final results, when I meet the assertion error, I remove that line of code, and it runs successfully, my meaning is that I can not achieve the performance which is in the original paper with running the code by myself in my own GPU machine, I train the EquiDock from the scratch on the dips dataset, the results of the dips are: `For  dips  method =  equidock ; num test files = 100  complex_rmsd_CA median/mean/std =  14.964256286621094 / 15.607631420575272  +-  7.329030992174103  complex_interface_rmsd CA median/mean/std =  11.410481452941895 / 12.017205838562273  +-  5.641359052105475`, it has lower results than the original paper and it is also lower than the model provided by the authors. I am not sure if you are having this problem.

and I also do not know why the receptor RMSD is always equal to 0 during the training stage. It also makes me confused.

**Comment by AxelGiottonini** [comment]
Hey ! If you trained EquiDock from scratch, then it is normal to obtain different results although their should be in the same magnitude order. As you are training a neural network with random initial parameters you would not expect the outputs to be equals between different models. So yes, I had similar results but I would not consider that as a problem :) 

For the receptor RMSD, you will always obtain a value of 0 because the model is just looking for a SE(3) transformation of the ligand. Thus, the receptor does not move from its initial position, that's why you obtain an RMSD of 0. If you pay attention to the article, the behavior is described as _"We design EQUIDOCK, a fast, end-to-end method for rigid body docking that directly predicts the SE(3) transformation to place one of the proteins (ligand) at the right location and orientation with respect to the second protein (receptor)."_

**Comment by onlyonewater** [comment]
hi, @AxelGiottonini, thanks for your detailed reply, yeah I got it, thanks, and I have no problem.

**Comment by onlyonewater** [comment]
and this is my results:
for dips:
![image](https://user-images.githubusercontent.com/41667733/224655898-4781e402-d3c7-45f4-8f74-4fb5656ee05d.png)

for db5.5:
![image](https://user-images.githubusercontent.com/41667733/224655951-b61ce70e-b118-4cf7-8937-8d56cb104090.png)

so, the results are all lower than the original paper. It can be seen that the results of db5 is more lower than the original paper.
